scripts/demangle.py

```diff
@@ -59,46 +82,110 @@ def demangle(func: str):
     #     # Demangling failed
     #     print("Demangling failed with status", status)
 
-for json_file_name in file_names:
-    file_name = json_file_name.split('.json')[0]
-    # Find file path
-    if len(glob.glob(proj_path+'/'+file_name+'.c*'))==0:
-        print('ERROR: File '+proj_path+'/'+file_name+'.c*'+" NOT found")
-        continue
-    file_path = glob.glob(proj_path+'/'+file_name+'.c*')[0]
-    # if file_name == 'augeas_api_fuzzer': continue
-    # open Json file and get filename + func name
-    with open(json_path+'/'+json_file_name, 'r') as f:
-        try:
-            data = json.load(f)
-        except:
+
+# get source code from docker
+def get_source_code(file_path: str, file_name: str, proj_name: str) -> Optional[str]:
+    # Create a container from the image
+    client = docker.from_env()
+    image = client.images.get("gcr.io/oss-fuzz/" + proj_name + ":latest")
+    container = client.containers.run(image, detach=True)
+    # check file_path, it may not the absolut path
+    if not "/" in file_path:
+        file_path = "/src/" + proj_name + "/" + file_path
+    try:
+        code_content = ""
+        for chunk in container.get_archive(file_path)[0]:
+            code_content += chunk.decode("utf-8")
+    except docker.errors.NotFound:
+        code_content = "ERROR"
+        warning(f"ERROR: File  {file_path}  NOT found\n")
+    # Stop and remove the container
+    container.stop()
+    container.remove()
+    return code_content
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Catch source code")
+    parser.add_argument(
+        "-n",
+        "--name",
+        type=str,
+        required=False,
+        default="coturn",
+        help="The project name to fetch",
+    )
```

Comment:
can you also put this part of the code in `Projects.postprocess()`? you can get the project name using `self.project` in the class.

Reply:
Yes, I also keep the demangle.py for debugging and testing.